### PR TITLE
[FIX] base: keep evaluation context unchanged

### PR DESCRIPTION
**Pull request by mart-e**
Before this commit, the evaluation context was forced with sudo(False)
as the call to _render_qweb_html was made with sudo(True). The sudo
was removed to prevent rendering a report on any object.

This was problematic for reports such as sale.order portal page where
the public user does not have access to the record but uses a custom
verification mechanism using an access_token field and then calling
_render with sudo.

This commit moves the sudo call to keep the _get_evaluation_context
with the same environment and allow to call _render with sudo

Add tests

### CODE REVIEW

**Comment by mart-e** [comment]
cc @william-andre should fix your issue in portal

**Comment by mart-e** [comment]
@robodoo r+

File: odoo/addons/base/models/ir_actions_report.py [hunk 1]
```diff
@@ -810,23 +810,23 @@ def _get_rendering_context_model(self):
 
     @api.model
     def _get_rendering_context(self, docids, data):
+        # access the report details with sudo() but evaluation context as current user
+        self_sudo = self.sudo()
+
         # If the report is using a custom model to render its html, we must use it.
         # Otherwise, fallback on the generic html rendering.
-        report_model = self._get_rendering_context_model()
+        report_model = self_sudo._get_rendering_context_model()
```
Comment (qsb-odoo) [inline odoo/addons/base/models/ir_actions_report.py:818]: Does passing sudo() has a future for version 13? Herewith is the attached link to the migration ticket. There is also one video attached to it demonstrating the refusal for accessing cost lines when the default website belongs to a different company than that of the production order's company.

https://www.odoo.com/web#id=2335874&action=333&active_id=70&model=project.task&view_type=form&cids=2&menu_id=4720

Comment (mart-e) [reply odoo/addons/base/models/ir_actions_report.py:818]: @qsb-odoo this PR is for 14.0, your ticket is for 13.0. We do not plan to backport this as it is part of a bigger refactoring of access rights that was done between 13 and 14.0

**Pull request closed without merging**